.github/workflows/ci.yml

```diff
@@ -35,21 +35,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: "7.4"
-          tools: composer:v2
-
-      - name: Get Composer cache directory
-        id: composer-cache
-        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-
-      - name: Setup Composer cache
-        uses: actions/cache@v4
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: php-7.4-composer-${{ hashFiles('**/composer.json') }}
-          restore-keys: php-7.4-composer-
 
-      - name: Install dependencies
-        run: composer install --prefer-dist --no-interaction --no-progress --optimize-autoloader
+      - name: Install PHP dependencies
+        uses: ramsey/composer-install@v3
```

Comment:
`composer-installer` is a high quality action: shell script only, includes caching.